src/Service/Modules/SalesPackageModule.cs

```diff
@@ -56,14 +66,35 @@ private object GetSalesPackages()
         private object GetSalesPackages(int pageNumber, int pageSize)
         {
             return this.Negotiate.WithModel(this.salesPackageService.GetAll(pageNumber, pageSize))
-                .WithMediaRangeModel("text/html", ApplicationSettings.Get).WithView("Index");
+                .WithMediaRangeModel("text/html", ApplicationSettings.Get)
+                .WithView("Index");
+        }
+
+        private object GetSalesPackages(int pageNumber, int pageSize, string sortBy, bool ascending)
+        {
+            return this.Negotiate
+                .WithModel(
+                    this.salesPackageService.GetAll(
+                        pageNumber,
+                        pageSize,
+                        this.GetSortExpressionOnProperty(sortBy),
+                        ascending))
+                .WithMediaRangeModel("text/html", ApplicationSettings.Get)
+                .WithView("Index");
         }
 
         private object GetSalesPackage(int id)
         {
             var result = this.salesPackageService.GetById(id);
-            return this.Negotiate.WithModel(result).WithMediaRangeModel("text/html", ApplicationSettings.Get)
+            return this.Negotiate.WithModel(result)
+                .WithMediaRangeModel("text/html", ApplicationSettings.Get)
                 .WithView("Index");
         }
+
+        private Expression<Func<SalesPackage, string>> GetSortExpressionOnProperty(string sortBy)
+        {
+            var param = Expression.Parameter(typeof(SalesPackage));
+            return Expression.Lambda<Func<SalesPackage, string>>(Expression.Property(param, sortBy), param);
```

Comment:
Think I'd have to debug this to figure out what it means!

Reply:
I'll put some documentation on it